### PR TITLE
new backup job design

### PR DESCRIPTION
**Pull request by gabedealmeida**

### CODE REVIEW

**Comment by super3** [comment]
@gabedealmeida Add number of files in the job

File: src/store/index.ts [hunk 1]
```diff
@@ -283,6 +284,15 @@ export const store = createStore<State>({
 			return directory;
 		},
 
+		async getFileCount(
+			{},
```
Comment (montyanderson) [inline src/store/index.ts:288]: Remove this?

**Comment by super3** [comment]
@gabedealmeida @montyanderson Resolve conflict?

File: index.ts [hunk 2]
```diff
@@ -171,6 +172,39 @@ const loadURL = serve({ directory: `${__dirname}/dist` });
...
+		async (event, { path }: { path: string }): Promise<number> => {
+			const getAllDirFilesCount = function (
+				dirPath: string,
+				count: number
+			) {
+				const files = fs.readdirSync(dirPath);
+
+				count = count || 0;
+
+				files.forEach(function (file) {
```
Comment (montyanderson) [inline index.ts:186]: can we refactor this to use a single `reduce()` function?

**Pull request merged**